### PR TITLE
Adding a Kanister function to delete CSI VolumeSnapshot

### PR DESCRIPTION
**Pull request by shlokchaudhari9**
## Change Overview

This PR consist of a new Kanister function to delete CSI VolumeSnapshot along with it's documentation

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [x] :world_map: Documentation
- [x] :robot: Test

## Issues

- #1126 

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by shlokchaudhari9** [comment]
The unit test in this PR depends on methods from restore PR. Following PRs need to be merged first.

File: pkg/function/delete_csi_snapshot_test.go
Comment (viveksinghggits) [inline pkg/function/delete_csi_snapshot_test.go:28]: import order is not correct. the order is 
```
standard packages

third party packages

kanister packages
```

File: pkg/function/delete_csi_snapshot_test.go
Comment (viveksinghggits) [inline pkg/function/delete_csi_snapshot_test.go:134]: instead of creating the context at places, let's create on and use that at other places.

Comment (shlokchaudhari9) [reply pkg/function/delete_csi_snapshot_test.go:135]: Done

**Comment by viveksinghggits** [comment]
LGTM

**Review by pavannd1 — COMMENTED** [review]
LGTM. Good to merge after merging restore.

**Comment by shlokchaudhari9** [comment]
@pavannd1 @viveksinghggits @PrasadG193 Need atleast one approval here

**Review by viveksinghggits — APPROVED** [review]

File: pkg/function/delete_csi_snapshot_test.go
Comment (viveksinghggits) [inline pkg/function/delete_csi_snapshot_test.go:32]: comments should start with the funciton/variable we are explaning.

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**
